include/revng/Pipeline/Container.h

```diff
@@ -55,13 +58,20 @@ class ContainerBase {
   /// not after(this)->enumerate().contains(Targets);
   virtual bool remove(const TargetsList &Targets) = 0;
 
+  virtual llvm::Error serialize(llvm::raw_ostream &OS) const = 0;
+
+  virtual llvm::Error deserialize(const llvm::MemoryBuffer &Buffer) = 0;
+
+  /// Must reset the state of the container to the just built state
+  virtual void clear() = 0;
```

Comment:
because if we are loading the entire state of a folder and a file is missing this conceptually means that the container should be default constructed. But we don't know how to default construct it because it is a virtual class and thus we need to be able to reset it.